rc/tools/clipboard.kak

```diff
@@ -0,0 +1,94 @@
+} str clipboardcmd_copy %sh{
+    for cmd in "xsel -i" "pbcopy" "wl-copy -p"; do
+        readonly program="${cmd%% *}"
+        if command -v "${program}" >/dev/null 2>&1; then
+            printf %s "${cmd}"
+            exit
+        fi
+    done
+}
+
+declare-option -docstring %{
+    The shell command which will paste the system clipboard into the current buffer
+} str clipboardcmd_paste %sh{
+    for cmd in "xsel -o 2>/dev/null" "pbpaste" "wl-paste -p"; do
+        readonly program="${cmd%% *}"
+        if command -v "${program}" >/dev/null 2>&1; then
+            printf %s "${cmd}"
+            exit
+        fi
+    done
+}
+
+define-command -params .. -docstring %{
+    clipboard-copy [<data>…]: copy the given data to the system clipboard
+
+    If no data is passed as argument to the command, the current selection is used
+    If multiple selections/data arguments are to be stored into the clipboard, they are joined with a newline character
+} clipboard-copy %{ evaluate-commands %sh{
+    join() {
+        # NOTE: this is a POSIX way of storing a newline
+        # character into $IFS without spawning a subshell
+        IFS='
+'
+        printf %s "$*"
+    }
+
+    if [ $# -eq 0 ]; then
+        eval "set -- $kak_quoted_selections"
+    fi
+
+    if ! join "$@" | eval "${kak_opt_clipboardcmd_copy}"; then
+        fail "Unable to copy data to the system clipboard"
+    fi
+} }
+
+define-command -docstring %{
+    Paste the contents of the system clipboard before the selection
+} clipboard-paste-before %{
+    execute-keys ! %opt{clipboardcmd_paste} <ret>
+}
+
+define-command -docstring %{
+    Paste the contents of the system clipboard after the selection
+} clipboard-paste-after %{
+    execute-keys <a-!> %opt{clipboardcmd_paste} <ret>
+}
+
+define-command -docstring %{
+    Replace the selection with the contents of the system clipboard
+} clipboard-replace %{
+    evaluate-commands -save-regs c %{
+        set-register c %sh{ eval "${kak_opt_clipboardcmd_paste}" }
+        execute-keys \"c R
+    }
+}
+
+define-command -docstring %{
+    Automatically copy the contents of the `c` register to the clipboard when it is modified
+} clipboard-enable-register %{
+    try clipboard-disable-register
+    hook -group clipboard-bind-register global RegisterModified c %{
+        clipboard-copy %reg{c}
+    }
+}
+
+define-command -docstring %{
+    Disable integration of the system clipboard with the `c` register
+} clipboard-disable-register %{
+    remove-hooks global clipboard-bind-register
+}
+
+# NOTE: Users who discover the command interactively might not want to
+# have to think about the before/after side effects, so assume they expect
+# pasting to always be performed after the selection
+alias global clipboard-paste clipboard-paste-after
+
+declare-user-mode clipboard-mode
+
+map global clipboard-mode -docstring "Copy selection to clipboard" y ": clipboard-copy<ret>"
```

Comment:
It's not strictly necessary, but people might expect to find "Cut selection to clipboard" here. I guess the implementation would be `: clipboard-copy; exec d<ret>` (since `exec` saves and restores the default register, `exec d` would delete the selection without disturbing any other registers).